internal/controller/streams.go

```diff
@@ -197,105 +164,225 @@ func (sm *streamManager) BootUpStreamAndReconnect(ctx context.Context, cancel co
 		return err
 	}
 	snapshotCompleted.Done()
+
+	<-ctx.Done()
+	return ctx.Err()
+}
+
+// StreamLogs handles the log stream.
+func (sm *streamManager) StreamLogs(ctx context.Context, cancel context.CancelFunc) error {
```

Comment:
`cancel` is not needed, so remove it from the signature:
```suggestion
func (sm *streamManager) StreamLogs(ctx context.Context) error {
```